nsv13/code/modules/power/stormdrive.dm

```diff
@@ -199,18 +195,28 @@ Takes  plasma and outputs superheated plasma and a shitload of radiation.
 	if(!in_range(src, usr) || !reactor)
 		return
 	if(href_list["rods_1"])
-		reactor.control_rod_state = RODS_RAISED
+		reactor.control_rod_percent = 0
 		message_admins("[key_name(usr)] has fully raised reactor control rods in [get_area(usr)] [ADMIN_JMP(usr)]")
 		reactor.update_icon()
 	if(href_list["rods_2"])
-		reactor.control_rod_state = RODS_HALFRAISED
+		reactor.control_rod_percent = 23.49
 		reactor.update_icon()
 	if(href_list["rods_3"])
-		reactor.control_rod_state = RODS_HALFLOWERED
+		reactor.control_rod_percent = 33.42
 		reactor.update_icon()
 	if(href_list["rods_4"])
-		reactor.control_rod_state = RODS_LOWERED
+		reactor.control_rod_percent = 100
 		reactor.update_icon()
+	if(href_list["rods_c"])
+		if(alert(usr, "Manually setting control rods positions should only be attempted by experienced nuclear engineers.", "EXTREME CAUTION ADVISED", "Continue", "Abort")=="Continue")
+			var/new_control_rod_percent = input(usr, "Set new control rod insertion depth percent", "Manual Control Rods Positional Control", reactor.control_rod_percent) as num|null
+			if(!usr.canUseTopic(src, BE_CLOSE))
+				return
+			new_control_rod_percent = CLAMP(new_control_rod_percent, 0, 100)
+			reactor.control_rod_percent = new_control_rod_percent
```

Comment:
```suggestion
			if (!isnull(new_control_rod_percent))
				new_control_rod_percent = CLAMP(new_control_rod_percent, 0, 100)
				reactor.control_rod_percent = new_control_rod_percent
```
If a player hits cancel on the percentage prompt, the control rod percentage gets set to null. This is approximately the worst possible thing, because it sets the icon of the reactor to display as if the rods are 100% inserted, but actually causes the reactor to not cool at _all._ Wrapping the assignment in an isnull() if statement should result in what I think is more intuitive behavior, which is retaining the current percentage on cancelling the prompt.

Reply:
tbh I wanted a TGUI solution rather than what I did.

Reply:
Should function correctly now.